types/activity/pub.ts

```diff
@@ -0,0 +1,105 @@
+import { Review } from '../review';
+import { Diff } from '../util';
+
+import { InsertableActivityItemBase } from './util';
+import { DiscussionActivityItemBase } from './discussion';
+import { ThreadActivityItemBase } from './thread';
+
+type PubActivityItemBase = InsertableActivityItemBase & {
+	pubId: string;
+	payload: {
+		pub: {
+			title: string;
+		};
+	};
+};
+
+export type PubCreatedActivityItem = PubActivityItemBase & {
+	kind: 'pub-created';
+};
+
+export type PubUpdatedActivityItem = PubActivityItemBase & {
+	kind: 'pub-updated';
+	payload: {
+		title?: Diff<string>;
+		doi?: Diff<null | string>;
+		attributions?: true;
+		draft?: true;
+	};
+};
+
+export type PubRemovedActivityItem = PubActivityItemBase & {
+	kind: 'pub-removed';
+};
+
+export type PubReleasedActivityItem = PubActivityItemBase & {
+	kind: 'pub-released';
+	payload: {
+		releaseId: string;
+	};
+};
+
+type PubEdgeActivityItemBase = PubActivityItemBase & {
+	payload: {
+		pubEdgeId: string;
+		target: {
```

Comment:
Not important to fix this now but after reviewing the play this should read:

```ts
target:
| {
        pub: {
            id: string;
            title: string;
            slug: string;
        };
  }
| {
        externalPublication: {
            id: string;
            title: string;
            url: string;
        };
  };
```
since we're guaranteed to have one or the other but not both.